bxsolana/provider/grpc.ts

```diff
@@ -61,22 +67,42 @@ import { BaseProvider } from "./base.js"
 import { Client } from "@grpc/grpc-js"
 import { CallMetadataOptions } from "@grpc/grpc-js/src/call-credentials"
 import config from "../../utils/config.js"
+import {WsProvider} from "./ws";
 
 export class GrpcProvider extends BaseProvider {
     private client: Service
     private grpcClient: Client
 
     constructor(address = `${MAINNET_API_GRPC_HOST}:${MAINNET_API_GRPC_PORT}`) {
+        console.log(address)
         super()
         const metaCallback = (options: CallMetadataOptions, cb: (err: Error | null, metadata?: grpc.Metadata) => void) => {
             const meta = new grpc.Metadata()
             meta.add("Authorization", config.AuthHeader)
             cb(null, meta)
         }
-        this.grpcClient = new Client(
-            address,
-            grpc.credentials.combineChannelCredentials(grpc.credentials.createSsl(), grpc.credentials.createFromMetadataGenerator(metaCallback))
-        )
+
+        let grpcClient: Client
+
+        if (process.env.API_ENV === "testnet") {
```

Comment:
`if` condition can be combined –– only need `process.env.API_ENV === "mainnet"` then everything can be in the `else`

Reply:
Actually, this specific class should not be aware of `process.env`. Pass in a separate variable along the lines of `useTls` like we do in the Go SDK